driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -21,116 +21,200 @@
+     *
+     * <p>Since integers are a subset of numbers, if a value is an integer,
+     * this does not imply that it is not a number, and vice-versa.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
     NumberExpression isNumberOr(NumberExpression other);
+
+    /**
+     * {@code this} value as an {@link IntegerExpression integer} if
+     * {@code this} is an integer, or the {@code other} integer value if
+     * {@code this} is null, or is missing, or is of any other non-integer type.
+     *
+     * <p>Since integers are a subset of numbers, if a value is an integer,
+     * this does not imply that it is not a number, and vice-versa.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
     IntegerExpression isIntegerOr(IntegerExpression other);
+
+    /**
+     * {@code this} value as a {@link BooleanExpression boolean} if
+     * {@code this} is a boolean, or the {@code other} boolean value if
+     * {@code this} is null, or is missing, or is of any other non-boolean type.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
     StringExpression isStringOr(StringExpression other);
+
+    /**
+     * {@code this} value as a {@link StringExpression string} if
+     * {@code this} is a string, or the {@code other} string value if
+     * {@code this} is null, or is missing, or is of any other non-string type.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
     DateExpression isDateOr(DateExpression other);
+
+
+    /**
+     * {@code this} value as a {@link ArrayExpression array} if
+     * {@code this} is an array, or the {@code other} array value if
+     * {@code this} is null, or is missing, or is of any other non-array type.
+     *
+     * <p>Warning: this operation does not guarantee type safety. While this
+     * operation is guaranteed to produce an array, the type of the elements of
+     * that array are not guaranteed by the API. The specification of a type by
+     * the user is an unchecked assertion that all elements are of that type,
+     * and that no element is null, is missing, or is of some other type. If the
+     * user cannot make such an assertion, some appropriate super-type should be
+     * chosen, and elements should be individually type-checked.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     * @param <T> the type of the elements of the resulting array.
+     */
     <T extends Expression> ArrayExpression<T> isArrayOr(ArrayExpression<T> other);
+
+    // TODO-END doc after Map merged, "record" and "schema objects" are decided
     <T extends DocumentExpression> T isDocumentOr(T other);
 
 
     /**
-     * server error if type cannot be converted to string (arrays, objects)
+     * The {@link StringExpression string} value corresponding to this value.
```

Comment:
corresponding -> representing?

That's what `Object.toString` uses, and it indeed looks like the right word. An arbitrary mapping from `Expression` to `StringExpression` establishes correspondence, but that does not necessary mean that the strings corresponding to values in this mapping represent the values in a way that `asString` is supposed to: even if there is no `parse` that will take the result of `asString` and produce the represented value, a human may look at the result of `asString` and understand what value it represents.

Reply:
Let's go with `The {@link StringExpression string} representation of {@code this} value`